Fix signing keys management for APT

davidcassany opened the pull request:
This commit fixes the management of the trusted keyring for apt
repositories. It creates a `trusted.gpg` keyring with the provided
signing keys so APT can check against that the configured repositories.

Fixes #1440

davidcassany commented:
@schaefi this PR fixes #1440 however it includes a couple of behavior changes that I'd like to discuss:

* The trusted keyring is created by using `gpg` command. This is a new dependency on the host for APT package manager if used with custom keys (`--signing-key` flag). I am not sure how should we treat that at packaging level. Probably adding gpg in `kiwi-packagemanager:apt` is enough. What do you think?

* The trusted keyring is not included into the resulting image, just used at buildtime, this differs from rpm based package manager (in rpm making this distinction in not obvious). Shall we reproduce a similar behavior? I'd say no, keep it simple and user can still include any custom keyring at config.sh.

schaefi commented:
@davidcassany  Thanks much for looking into this

>I am not sure how should we treat that at packaging level. Probably adding gpg in kiwi->packagemanager:apt is enough. What do you think? 

The code is now invoking gpg which I think is worth at least a recommends in the spec file like we do it for dnf and other packages that are used only on demand. Also gpg should be added to the box descriptions that are Debian based (the ubuntu box only atm)

>The trusted keyring is not included into the resulting image, just used at buildtime, this differs from rpm based package manager (in rpm making this distinction in not obvious). Shall we reproduce a similar behavior? I'd say no, keep it simple and user can still include any custom keyring at config.sh.

I agree let's keep this simple. Do you think it's worth to print a warning on the console that the gpg signature for the repos will not be permanent in the image ? What also came to my mind is; Do we have a chapter about signing-keys for repos and packages in the documentation ? If yes we should add a behavior comment in this case if not I think it's worth to have one (new issue)

davidcassany commented:
> The code is now invoking gpg which I think is worth at least a recommends in the spec file like we do it for dnf and other packages that are used only on demand.

Yes this sounds reasonable to me.

> I agree let's keep this simple. Do you think it's worth to print a warning on the console that the gpg signature for the repos will not be permanent in the image ?

Yes that might be a good idea, we used to have a warning to let users know keys were not being used in bootstrap phase, probably we should warn about the keyring not being included into the resulting image.

> What also came to my mind is; Do we have a chapter about signing-keys for repos and packages in the documentation ? If yes we should add a behavior comment in this case if not I think it's worth to have one (new issue)

I don't think we have a section for that. But this is an interesting topic, moreover I believe each package manager might behave slightly different.

davidcassany commented:
Ok, now I have added a warning to advise users the custom keyring is only present at buildtime and I also added the gpg tool as a recommended dependency.

schaefi commented:
@davidcassany  I think the commits can be merged, can't they ?

davidcassany commented:
> @davidcassany I think the commits can be merged, can't they ?

yes, I am merging
